packages/identity-ans/src/identity-ans.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { Connection, PublicKey } from '@solana/web3.js';
+import { ANSIdentityResolver } from './identity-ans';
+
+const RPC_URL = 'https://api.mainnet-beta.solana.com';
+const connection = new Connection(RPC_URL);
+
+describe('ans tests', () => {
+  it('should perform direct lookup', async () => {
```

Comment:
Getting error when run the test, any ideas?

```
Error: 410 Gone:  {"jsonrpc":"2.0","error":{"code": 410, "message":"The RPC call or parameters have been disabled."}, "id": "113f39bf-59ef-46ac-8d21-88f6668764e2" } 
    at ANSIdentityResolver.resolve
```

Reply:
This looks like a public RPC problem, they might have disabled getProgramAccounts

I ran into it before, try using a different/private rpc

https://stackoverflow.com/questions/72302043/error-410-gone-the-rpc-call-or-parameters-have-been-disabled

Reply:
Got it! Running using private rpc produced different error, any ideas what can be wrong?

```
Error: Attempt to access memory outside buffer bounds
    at ANSIdentityResolver.resolve (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.ts:61:15)
```

Reply:
@tonivlad just run the test as it is with private RPC url

Reply:
> Got it! Running using private rpc produced different error, any ideas what can be wrong?
> 
> ```
> Error: Attempt to access memory outside buffer bounds
>     at ANSIdentityResolver.resolve (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.ts:61:15)
> ```

hi, can you please double check if you were on mainnet or devnet?

Reply:
The error below happens on mainnet
```
Error: Attempt to access memory outside buffer bounds
    at ANSIdentityResolver.resolve (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.ts:61:15)
```
On devnet there's no error above but both tests still fail

```
Error: expect(received).toStrictEqual(expected) // deep equality

- Expected  - 2
+ Received  + 2

  Object {
    "additionals": Object {
-     "displayName": StringContaining "miester.poor",
+     "displayName": "a2.poor     ",
    },
    "address": "2EGGxj2qbNAJNgLCPKca8sxZYetyTjnoRspTPjzN2D67",
-   "name": StringContaining "miester.poor",
+   "name": "a2.poor     ",
    "type": "ANS",
  }
<Click to see difference>


    at Object.<anonymous> (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.spec.ts:16:22)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

Error: expect(received).toStrictEqual(expected) // deep equality

Expected: {"additionals": {"displayName": "miester.poor"}, "address": "2EGGxj2qbNAJNgLCPKca8sxZYetyTjnoRspTPjzN2D67", "name": "miester.poor", "type": "ANS"}
Received: null
<Click to see difference>


    at Object.<anonymous> (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.spec.ts:30:22)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Reply:
@crypt0miester thanks! I see you bumped `@onsol/tldparser` version, please also commit the updated `yarn.lock` file
@tonivlad @crypt0miester the test however still fails to run for me after the changes, here're are the errors
mainnet, `should perform direct lookup` test case failed:
```
Error: Cannot read properties of undefined (reading 'data')

    at ANSIdentityResolver.resolve (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.ts:61:15)
```
devnet, both test cases failed:

```
Error: expect(received).toStrictEqual(expected) // deep equality

- Expected  - 2
+ Received  + 2

  Object {
    "additionals": Object {
-     "displayName": StringContaining "miester.poor",
+     "displayName": "a2.poor     ",
    },
    "address": "2EGGxj2qbNAJNgLCPKca8sxZYetyTjnoRspTPjzN2D67",
-   "name": StringContaining "miester.poor",
+   "name": "a2.poor     ",
    "type": "ANS",
  }
<Click to see difference>

    at Object.<anonymous> (/Users/tsmbl/projects/dialect/sdk/packages/identity-ans/src/identity-ans.spec.ts:16:22)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

```
Please, double-check that tests pass for you

Reply:
our tests are run in mainnet, please use a mainnet rpc. 

sure, will fix the yarn.lock.

Reply:
btw, what node version are you on?

Reply:
@crypt0miester `v16.15.0`

Reply:
@tsmbl fixed yarn.lock. did the test on mainnet pass?

Reply:
@crypt0miester, ty
no, still have the error

```
    TypeError: Cannot read properties of undefined (reading 'data')
        at TldParser.<anonymous> (/Users/tsmbl/projects/dialect/sdk/node_modules/@onsol/tldparser/src/parsers.ts:185:40)
```

does it pass for you?

Reply:
So, 

we will upgrade our programs today to implement main/favorite domains and upgrade our tldparser lib to take this into account

the identity lib took the first domain and the test checked for that, but some RPCs also replied with a user's subdomains (we use that for DNS records) and that's why it failed (weird behaviour if you ask me)

the upgrade will fix this issue

I will come back to you here when we have everything done & ready

btw, thank you for everything